python/veles/messages/fields.py

```diff
@@ -67,54 +71,70 @@ def validate(self, value):
                 self.name, self.maximum))
         return value
 
+    def cpp_type(self):
+        return 'int64_t'
+
 
 class Boolean(Field):
     def validate(self, value):
-        super().validate(value)
+        super(Boolean, self).validate(value)
         if value is None:
             return
         if not isinstance(value, bool):
             raise ValueError('Attribute {} has to be bool type.'.format(
                 self.name))
         return value
 
+    def cpp_type(self):
+        return 'bool'
+
 
 class Float(Field):
     def validate(self, value):
-        super().validate(value)
+        super(Float, self).validate(value)
         if value is None:
             return
         if not isinstance(value, float):
             raise ValueError('Attribute {} has to be float type.'.format(
                 self.name))
         return value
 
+    def cpp_type(self):
+        return 'double'
+
 
 class String(Field):
     def validate(self, value):
-        super().validate(value)
+        super(String, self).validate(value)
         if value is None:
             return
         if not isinstance(value, str):
             raise ValueError('Attribute {} has to be str type.'.format(
                 self.name))
         return value
 
+    def cpp_type(self):
+        return 'std::string'
+
 
 class Binary(Field):
     def validate(self, value):
-        super().validate(value)
+        super(Binary, self).validate(value)
         if value is None:
             return
         if not isinstance(value, bytes):
             raise ValueError('Attribute {} has to be str type.'.format(
                 self.name))
         return value
 
+    def cpp_type(self):
```

Comment:
By default msgpack-c doesn't treat vector of uint8_t as bin type - https://github.com/msgpack/msgpack-c/wiki/v2_0_cpp_adaptor. We would need to add custom handling for that (not that it would be hard).